src/core/SaleorSDK.ts

```diff
@@ -1,14 +1,17 @@
-import { ApolloClient, NormalizedCacheObject } from "@apollo/client";
-import { auth, AuthSDK } from "./auth";
+import { auth } from "./auth";
+import { createSaleorClient } from "../apollo";
+import { Core, SaleorSDKConfig } from "./types";
 
-export interface Core {
-  auth: AuthSDK;
-}
+export const SaleorSDK = ({ apiUrl, channel }: SaleorSDKConfig): Core => {
+  let _channel = channel;
+  const changeChannel = (channel: string): string => {
+    _channel = channel;
+    return _channel;
+  };
 
-export const SaleorSDK = (
-  client: ApolloClient<NormalizedCacheObject>
-): Core => {
-  const authSDK = auth(client);
+  const apolloClient = createSaleorClient(apiUrl);
+  const coreInternals = { apolloClient, channel };
+  const authSDK = auth(coreInternals);
 
-  return { auth: authSDK };
+  return { auth: authSDK, _internal: { ...coreInternals, changeChannel } };
```

Comment:
`changeChannel` and `channel` should be exposed outside `_internal` as they should be public and freely accessed. I'd also vote for `setChannel`, it's shorter and more standard name for setters. Putting them under `config` seems like a good idea. Example:
```suggestion
  return { auth: authSDK, config: { channel, setChannel },  _internal: { apolloClient } };
```